include/ck/host_utility/stream_utility.hpp

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2023, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include <hip/hip_runtime.h>
+
+#include "ck/stream_config.hpp"
+#include "ck/host_utility/hip_check_error.hpp"
+
+static int getAvailableComputeUnitCount(const StreamConfig& stream_config)
+{
+    constexpr int MAX_MASK_DWORDS = 64;
+
+    // assume at most 64*32 = 2048 CUs
+    uint32_t cuMask[MAX_MASK_DWORDS];
+
+    for(int i = 0; i < MAX_MASK_DWORDS; i++)
+        cuMask[i] = 0;
+
+    auto countSetBits = [](uint32_t dword) {
+        int count = 0;
+
+        while(dword != 0)
+        {
+            if(dword & 0x1)
+                count++;
+
+            dword = dword >> 1;
+        };
+
+        return (count);
+    };
+
+    hip_check_error(hipExtStreamGetCUMask(stream_config.stream_id_, MAX_MASK_DWORDS, &cuMask[0]));
```

Comment:
&cuMask[0] => cuMask

Reply:
Why not just getting  the `multiProcessorCount` property from the device ? https://docs.amd.com/bundle/HIP-API-Guide-v5.4.1/page/a01151.html#add8d9d2ad52aece9fd1dbe25c18d9d57

Reply:
> Why not just getting the `multiProcessorCount` property from the device ? https://docs.amd.com/bundle/HIP-API-Guide-v5.4.1/page/a01151.html#add8d9d2ad52aece9fd1dbe25c18d9d57

Cause the CUs usable by a stream can be masked by the `hipExtStreamCreateWithCUMask()`,  multiple streams can created on the same device,  we should concern about the stream, rather the device.